cs-jClouds/src/main/java/io/cloudslang/content/jclouds/actions/signature/ComputeSignatureV4.java

```diff
@@ -0,0 +1,103 @@
+     * @return A map, with strings as keys and values, that contains: outcome of the action, returnCode of the operation
+     * or failure message, the exception if there is one, signature value and authorization header value
+     */
+    @Action(name = "Compute Signature V4",
+            outputs = {
+                    @Output(Outputs.RETURN_CODE),
+                    @Output(Outputs.RETURN_RESULT),
+                    @Output(Outputs.EXCEPTION),
+                    @Output(Constants.AWSParams.SIGNATURE_RESULT),
+                    @Output(Constants.AWSParams.AUTHORIZATION_HEADER_RESULT)
+            },
+            responses = {
+                    @Response(text = Outputs.SUCCESS, field = Outputs.RETURN_CODE, value = Outputs.SUCCESS_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.FAILURE_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR, isOnFail = true)
+            })
+    public Map<String, String> computeSignature(@Param(value = Inputs.CommonInputs.ENDPOINT) String endpoint,
+                                                @Param(value = Inputs.CommonInputs.IDENTITY, required = true) String identity,
+                                                @Param(value = Inputs.CommonInputs.CREDENTIAL, required = true, encrypted = true)
+                                                        String credential,
+
+                                                @Param(value = Inputs.CustomInputs.AMAZON_API) String amazonApi,
+                                                @Param(value = Inputs.CustomInputs.URI) String uri,
+                                                @Param(value = Inputs.CustomInputs.HTTP_VERB) String httpVerb,
+                                                @Param(value = Inputs.CustomInputs.PAYLOAD_HASH) String payloadHash,
+                                                @Param(value = Inputs.CustomInputs.DATE) String date,
+                                                @Param(value = Inputs.CustomInputs.HEADERS) String headers,
+                                                @Param(value = Inputs.CustomInputs.QUERY_PARAMS) String queryParams,
+
+                                                @Param(value = Inputs.AWSApiInputs.SECURITY_TOKEN) String securityToken) {
+
+        Map<String, String> headersMap = InputsUtil.getHeadersMap(headers);
+        Map<String, String> queryParamsMap = InputsUtil.getQueryParamsMap(queryParams);
+
+        try {
+            AuthorizationHeader authorizationHeader = new AmazonSignatureService().signRequestHeaders(endpoint, identity,
+                    credential, amazonApi, uri, httpVerb, payloadHash, securityToken, date, headersMap, queryParamsMap);
```

Comment:
@gituser173 No... the intention here is to have this operation that can be used as stand alone... furthermore it has to be the same as it is in AFL ... that means the inputs/outputs as well